### PR TITLE
feat: Production Plan: add a production lead time setting and use it to set planned start dates on the plan from delivery dates

### PR DESCRIPTION
**Pull request by nameduser0**
fixes: #26465

no-docs

### CODE REVIEW

**Comment by nameduser0** [comment]
Unrelated test failure

**Comment by nameduser0** [comment]
@rmehta @umairsy As usual, valuable contributions ignored and marked stale

**Comment by nameduser0** [comment]
😆 [https://erpnext.org/contribute](https://erpnext.org/contribute)

File: erpnext/manufacturing/doctype/manufacturing_settings/manufacturing_settings.json [hunk 1]
```diff
@@ -175,13 +176,20 @@
    "fieldname": "add_corrective_operation_cost_in_finished_good_valuation",
    "fieldtype": "Check",
    "label": "Add Corrective Operation Cost in Finished Good Valuation"
+  },
+  {
+   "default": "10",
+   "description": "Subtracted from Sales Order and Material Request delivery dates for Production Plan start date",
+   "fieldname": "production_lead_time_days",
```
Comment (ankush) [inline erpnext/manufacturing/doctype/manufacturing_settings/manufacturing_settings.json:183]: A configuration like this on the BOM document would be better? Single lead time value can't apply to all items for most businesses.  

Also the linked issue you mentioned is fixed now, so Work Orders now respect set dates from ProdPlan.

https://github.com/frappe/erpnext/pull/27031

Comment (nameduser0) [reply erpnext/manufacturing/doctype/manufacturing_settings/manufacturing_settings.json:183]: I would suggest the BOM should have enough `Operation` information to work out a lead time if you wanted to. I'd suggest using this as a default. I don't have time to rework or improve this.

**Pull request closed without merging**